### ISSUE TITLE
Unable to query vault secret from HCL2

### ISSUE THREAD

**Issue by james-emerton** [hashicorp/packer#10494]
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

I'm trying to read a value from Vault KV v2 and Packer responds with a 400 error. I am able to access the secret using both the `vault` CLI and `curl`.

#### Reproduction Steps

With the following snippet in a Packer configuration:

```
locals {
   secret = vault("secret/data/foo", "password")
}
```

Executing Packer results in:

```
$ packer build .
Error: Error in function call

  on build.pkr.hcl line 2:
  (source code not available)

Call to function "vault" failed: Error reading vault secret: Error making API
request.

URL: GET https://vault-server/v1/secret/data/foo
Code: 400. Errors:

* error performing token check: failed to look up namespace from the token: no
namespace.
```

Yet I can successfully query exactly that URL with curl:

```
$ curl -H "X-Vault-Token: $VAULT_TOKEN" https://vault-server/v1/secret/data/foo
{"request_id":...}
```

### Versions

packer: 1.6.6
vault server (oss): 1.6.1

### Operating system and Environment details

MacOS 10.15.7

**Comment by SwampDragons** [comment]
This looks similar to an issue that came up this summer on our discuss forums: https://discuss.hashicorp.com/t/error-performing-token-check-failed-to-look-up-namespace-from-the-token-no-namespace/10746/9

Does any of the conversation there help?

**Comment by james-emerton** [comment]
Oh dear. I've wrapped packer in a `Makefile` which reads `~/.vault-token` into `$VAULT_TOKEN` and I did it wrong. I even thought I checked for this but my variable contained a shell expression which appeared to work. 

I had noticed in the [PR for this feature](https://github.com/jsmcnair/packer/blob/78c280b1b7245af27acf1a0b5d918398b2a12b25/common/template/funcs.go#L30) that the presence of `$VAULT_TOKEN` is tested explicitly, but my understanding is that the underlying library should also accept `~/.vault-token`. Would it be better to omit the check for the env var and report any failure from the underlying library?
